### PR TITLE
fix(responses): drain streaming chunks in background task handler

### LINKED ISSUE #1945: Responses API: background=true + stream=true fails with 'Unexpected response type'
`background: true` combined with `stream: true` currently fails background tasks with `"Unexpected response type"`.

## Summary

In the `create_response` background branch, the spawned task waits for exactly one message from `bg_rx.recv().await` and treats only `Some(Response::Done(_))` as success.

If the request is streaming, the model pipeline can emit `Response::Chunk(_)` first. That variant is not handled in the match and falls into the default `_` arm, which marks the task as failed with `unknown_error: Unexpected response type`.

## Affected code

- `mistralrs-server-core/src/responses.rs` background path in `create_response`
- Match on first received response:
  - handles `Some(Response::Done(_))`
  - falls through to `_ => mark_failed("Unexpected response type")`

## Logic proof

1. Request is accepted with `background=true` and `stream=true`.
2. Background worker does `match bg_rx.recv().await` once.
3. If first event is `Response::Chunk(_)`, it is unmatched.
4. `_` arm executes and task is marked failed.

So this is not just “background ignores streaming”; it actively fails on streamed chunk responses.

## Expected behavior

`background=true` and `stream=true` should not fail with `Unexpected response type` purely because chunk events are emitted before terminal completion.

### PR DESCRIPTION
**Pull request by glaziermag**
> [!NOTE]
> Agent 4 A100 validation update (2026-05-13 UTC): classification `ACTUAL`, feasibility `FEASIBLE_NOW`. On A100 base `2d4ba4f16f61e5e18be085d0dd137bc95cba038a`, the exact `/v1/responses` request with `background=true` and `stream=true` failed when polled with `unknown_error: Unexpected response type`. On PR head `61219e17256b67920e56d630a8c63b89b1e81be2`, the same request completed with `output_text:"4"` and `background:true`. Controls passed: `background=true, stream=false`, `background=false, stream=true`, and normal `/v1/chat/completions`; server log scan for `error|panic|channel_closed|failed` had no matches. “Fixes #1945” is honest for this path.

---

## Fix: Force Non-Streaming for Background Request Generation

This PR forces `stream=false` for background tasks inside the Responses API handler (`/v1/responses` with `background=true`) to prevent the generator from hanging indefinitely while waiting for an HTTP stream that does not exist.

### Root Cause

When `background=true` is set in a Responses API request, the handler spawns a background task that expects a terminal `Response::Done`. If the upstream request includes `stream=true`, the engine can send chunk events instead of the non-streaming terminal shape expected by that background task.

### Fix

1. Force `oairequest.stream = Some(false)` before `parse_openresponses_request` in the `if background {` branch.
2. Drain any stray `Response::Chunk` events and only complete the task on `Response::Done`.
3. If the response channel closes before a terminal message, mark the background task failed instead of leaving it stuck in progress.

### Runtime Validation (GCP g2-standard-8, L4 GPU, Qwen2.5-0.5B Q4_K_M GGUF)

**Branch head tested**: `bd8ad18517e8c1e10f8fe4d6dcd6528d8afa73bf`  
**Hardware**: GCP `g2-standard-8`, 1x NVIDIA L4, driver `580.126.09`, CUDA 12.9

**Build:**
```bash
cargo build --release --features cuda -p mistralrs-server
```
Result: build completed successfully on the L4 VM.

**Server:**
```bash
./target/release/mistralrs-server --port 18082 gguf \
  -m Qwen/Qwen2.5-0.5B-Instruct-GGUF \
  -f qwen2.5-0.5b-instruct-q4_k_m.gguf
```
Server log included:
```text
Model loaded.
git revision: bd8ad18517e8c1e10f8fe4d6dcd6528d8afa73bf
Dummy run completed in 7.515720829s.
OpenAI-compatible server listening on http://0.0.0.0:18082.
```

**Exact bug-trigger request: `background=true` + `stream=true` via `/v1/responses`:**
```bash
curl -sS -X POST http://127.0.0.1:18082/v1/responses \
  -H "Content-Type: application/json" \
  -d '{"model":"default","input":"What is 2+2? Answer with just the number.","background":true,"stream":true}'
```
Initial response:
```json
{"id":"resp_ffa42766-f0cc-4935-8c72-8c3c355ab22d","object":"response","created_at":1777337731,"model":"default","status":"queued","output":[],"metadata":null}
```

Poll command:
```bash
curl -sS http://127.0.0.1:18082/v1/responses/resp_ffa42766-f0cc-4935-8c72-8c3c355ab22d
```
Final poll response:
```json
{"id":"resp_ffa42766-f0cc-4935-8c72-8c3c355ab22d","object":"response","created_at":1777337731,"completed_at":1777337731,"model":"Qwen/Qwen2.5-0.5B-Instruct-GGUF","status":"completed","output":[{"type":"message","id":"msg_d24e9970-4040-4fd9-b0b4-d2c16c986fa5","role":"assistant","content":[{"type":"output_text","text":"4"}],"status":"completed"}],"output_text":"4","usage":{"input_tokens":42,"output_tokens":2,"total_tokens":44},"background":true}
```

Server log scan:
```bash
grep -Ei "error|panic|channel_closed|failed" /tmp/mistral_responses_1777337716.log
```
Result: no matches.

This validates the intended `/v1/responses` background + streaming request path on the PR head.

### A100 validation (2026-05-01)

Additional validation on GCP Spot `a2-highgpu-1g` in `[redacted-region]` with 1x NVIDIA A100-SXM4-40GB, driver `580.126.20`, branch head `61219e17256b67920e56d630a8c63b89b1e81be2`.

```bash
cargo build --release --features cuda -p mistralrs-server
./target/release/mistralrs-server --port 18082 gguf \
  -m Qwen/Qwen2.5-0.5B-Instruct-GGUF \
  -f qwen2.5-0.5b-instruct-q4_k_m.gguf
```

Result: build passed. The exact `/v1/responses` bug shape with `background=true` and `stream=true` returned an initial queued response, then completed when polled with `output_text":"4"` and `background":true`. A normal `/v1/chat/completions` smoke also returned `OK`. Server log scan for `error|panic|channel_closed|failed` had no matches.

### CODE REVIEW

**Comment by glaziermag** [comment]
**Housekeeping note:** This branch currently bundles the CI fix from #2115 (`.typos.toml`, `openapi_doc.rs`, `distributed/layers.rs`). Once #2115 is merged, this branch will need a rebase onto updated `master` to drop the duplicate CI fix commit and resolve the resulting conflicts.

**Comment by glaziermag** [comment]
Agent 6 follow-up on existing A100 validation: this remains valid. Classification: `ACTUAL`; feasibility: `FEASIBLE_NOW`. The recorded A100 before/after evidence reproduces the `/v1/responses` `background=true` + `stream=true` failure on base and shows current head completing successfully, so `Fixes #1945` is honest for that path. Recommendation: keep open for review/merge.

**Comment by glaziermag** [comment]
Wave 1 evidence bundle index:

PR: EricLBuehler/mistral.rs#2114
Linked issue: #1945
Base SHA: `2d4ba4f16f61e5e18be085d0dd137bc95cba038a`
Current PR-head SHA: `61219e17256b67920e56d630a8c63b89b1e81be2`
Fixed-head SHA, if changed: N/A

Exact server command recorded in the PR body:
```bash
cargo build --release --features cuda -p mistralrs-server
./target/release/mistralrs-server --port 18082 gguf \
  -m Qwen/Qwen2.5-0.5B-Instruct-GGUF \
  -f qwen2.5-0.5b-instruct-q4_k_m.gguf
```
The exact client path was `/v1/responses` with `background=true` and `stream=true`, then polling the queued response.

Environment: GCP Spot `a2-highgpu-1g`, zone `[redacted-region]`, 1x NVIDIA A100-SXM4-40GB, driver `580.126.20`, CUDA from `nvidia-smi` `13.0`, Rust/Cargo from PR-body A100 run, model `Qwen/Qwen2.5-0.5B-Instruct-GGUF` / `qwen2.5-0.5b-instruct-q4_k_m.gguf`.
A100 category: `A100_GPU_REQUIRED`.

Base result: exact `/v1/responses` request with `background=true` and `stream=true` failed when polled with `unknown_error: Unexpected response type`.
Current PR-head result: same request completed with `output_text: "4"` and `background: true`.
Tests added/changed: Responses API background streaming handler behavior.
Tests passed: A100 server/client runtime path; controls for `background=true, stream=false`, `background=false, stream=true`, and normal `/v1/chat/completions`; server log scan had no `error|panic|channel_closed|failed` matches.
Side-effect controls: non-background streaming and background non-streaming controls passed; normal chat completions smoke passed.
Raw logs/artifacts: existing validation comment https://github.com/EricLBuehler/mistral.rs/pull/2114#issuecomment-4455524590 plus PR-body server/client excerpts. No separate raw log file is attached in this comment.
Remaining risks: scoped to `/v1/responses` with `background=true` and `stream=true`; broader Responses API behavior is not claimed.
Can say “Fixes #issue”: yes for #1945 on this path.
Safe wording: “Fixes #1945 for `/v1/responses` with `background=true` and `stream=true`.”
Readiness status: `ready-now` if existing A100 comments/PR-body excerpts are accepted as artifacts; otherwise standalone raw server/client logs remain to attach.

**Comment by glaziermag** [comment]
Ready for maintainer review. Evidence attached shows the original/base failure and current-head pass under the scoped issue conditions. The PR may use the `Fixes #...` wording already present in the body.

**Comment by glaziermag** [comment]
Runtime validation of the fix on the current head (local merge with master):

**Environment**: macOS arm64 (8 GB, CPU-only), dev-opt build (`debug=0`), Rust 1.97.0. Model: `ggml-org/gemma-3-1b-it-GGUF` `Q4_K_M` (sha256 `8ccc5cd1f1b36025…`), served via `mistralrs serve -p 18923 text -m . --format gguf -f gemma-3-1b-it-Q4_K_M.gguf --max-seq-len 512`.

**Request** (the exact #1945 failure shape):

```bash
curl -s http://localhost:18923/v1/responses -H "Content-Type: application/json" \
  -d '{"model":"default","input":"Say hello.","background":true,"stream":true,"store":true,"max_output_tokens":8}'
# → 200 {"id":"resp_dc8fc0b0-…","status":"queued", …}

curl -s http://localhost:18923/v1/responses/resp_dc8fc0b0-…   # ~1s later
# → "status":"completed", well-formed assistant message, usage {"input_tokens":18,"output_tokens":1,"total_tokens":19}
```

No `unknown_error` / "Unexpected response type"; the background task reached a terminal state and the stored response was retrievable. The empty `output_text` reflects a separate Gemma3-GGUF logits issue noted on #1964 (reproducible without `background`/`stream`), orthogonal to the channel handling validated here.

Disclosure: one local, uncommitted harness tweak was required to run on this 8 GB host — an env override for the CPU available-memory probe (`sysinfo` reported 0 MB available). It affects only device-map admission, not the responses path under test.
